src/main/java/com/cburch/logisim/fpga/hdlgenerator/WithSelectHDLGenerator.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Logisim-evolution - digital logic design tool and simulator
+ * Copyright by the Logisim-evolution developers
+ *
+ * https://github.com/logisim-evolution/
+ *
+ * This is free software released under GNU GPLv3 license
+ */
+package com.cburch.logisim.fpga.hdlgenerator;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.cburch.logisim.util.LineBuffer;
+
+public class WithSelectHDLGenerator {
+  
+  private final Map<Integer,Integer> myCases;
+  private final String regName;
+  private final String sourceSignal;
+  private final int nrOfSourceBits;
+  private final String destinationSignal;
+  private final int nrOfDestinationBits;
+  
+  public static final int OTHERS_INDEX = -1;
+  
+  public WithSelectHDLGenerator(String componentName, String sourceSignal, int nrOfSourceBits, 
+      String destinationSignal, int nrOfDestinationBits) {
+    myCases = new HashMap<>();
+    regName = LineBuffer.format("s_{{1}}_reg",componentName);
+    this.sourceSignal = sourceSignal;
+    this.nrOfSourceBits = nrOfSourceBits;
+    this.destinationSignal = destinationSignal;
+    this.nrOfDestinationBits = nrOfDestinationBits;
+  }
+  
+  private int binairyStringToInd(String binairyValue) {
+    var result = 0;
+    for (var charIndex = 0; charIndex < binairyValue.length(); charIndex++) {
+      final var character = binairyValue.charAt(charIndex) - '0';
+      if ((character < 0) || (character > 1)) throw new NumberFormatException("Invalid binairy value in WithSelectHDLGenerator");
+      result *= 2;
+      result += character;
+    }
+    return result;
+  }
+  
+  public void add(int selectValue, int assignValue) {
+    myCases.put(selectValue, assignValue);
+  }
+  
+  public void add(int selectValue, String binairyAssignValue) {
+    myCases.put(selectValue, binairyStringToInd(binairyAssignValue));
+  }
+  
+  public void add(String binairyselectValue, String binairyAssignValue) {
+    myCases.put(binairyStringToInd(binairyselectValue), binairyStringToInd(binairyAssignValue));
+  }
+  
+  public void add(String binairyselectValue, int assignValue) {
```

Comment:
Is it me, or there's something wrong with name `binairyselectValue`. First `select` should be capitalized, but that "binairy" does not seem like correct work?

Reply:
done